omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/UserController2_0Test.java

```diff
@@ -274,13 +285,111 @@ public void updateUser_shouldUpdateTheUserPassword() throws Exception {
 		assertNotNull(exception);
 		assertEquals("Invalid username and/or password: " + username, exception.getMessage());
 		
-		handle(newPostRequest(getURI() + "/" + user.getUuid(), "{\"password\":\"" + newPassword + "\"}"));
+		handle(newPostRequest(getPasswordURI() + "/" + user.getUuid(), "{\"password\":\"" + newPassword + "\"}"));
 		Context.logout();
 		
 		Context.authenticate(username, newPassword);
 		assertEquals(user, Context.getAuthenticatedUser());
 	}
 	
+	@Test
+	public void testChangeUsersOwnPassword() throws Exception {
+		setUpUser("butch");
+		
+		String oldPassword = "SomeOtherPassword123";
+		String newPassword = "newPassword9";
+		
+		MockHttpServletResponse response = handle(newPostRequest(getPasswordURI(), "{\"password\":\""
+		        + newPassword + "\"" + "," + "\"oldPassword\":\"" + oldPassword + "\"}"));
+		assertEquals(201, response.getStatus());
+	}
+	
+	@Test
+	public void testChangeUsersOwnPasswordWithOutAuthentication() throws Exception {
+		Context.logout();
+		String oldPassword = "SomeOtherPassword123";
+		String newPassword = "newPassword9";
+		
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("You are not authenticated to change password");
+		
+		handle(newPostRequest(getPasswordURI(), "{\"password\":\""
+				+ newPassword + "\"" + "," + "\"oldPassword\":\"" + oldPassword + "\"}"));
+	}
+	@Test
+	public void testChangeUsersOwnPasswordWithIncorrectOldPassword() throws Exception {
+		setUpUser("butch");
+		
+		String wrongOldPassword = "WrongPassword";
+		String newPassword = "newPassword9";
+		
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("old.password.not.correct");
+		
+		handle(newPostRequest(getPasswordURI(), "{\"password\":\"" + newPassword + "\"" + ","
+		        + "\"oldPassword\":\"" + wrongOldPassword + "\"}"));
+		
+	}
+	
+	@Test
+	public void testUserChangeOtherUsersPassword() throws Exception {
+		User authenticatedUser = setUpUser("daemon");
+		
+		Role role = new Role("Privileged Role");
+		role.addPrivilege(new Privilege(PrivilegeConstants.EDIT_USER_PASSWORDS));
+		authenticatedUser.addRole(role);
+		
+		String newPassword = "newPassword9";
+		
+		MockHttpServletResponse response = handle(newPostRequest(getPasswordURI() + "/" + getUuid(), "{\"password\":\""
+		        + newPassword + "\"}"));
+		
+		assertEquals(200, response.getStatus());
```

Comment:
I understand that's why it's happening. I'm just saying that for a client this doesn't make sense. (Specifically, changing a password should not return "Created"). Ideally we'd fix this but it's not a priority for me.